doc/modules/ensemble.rst

```diff
@@ -892,3 +892,95 @@ averaged.
  .. [HTF2009] T. Hastie, R. Tibshirani and J. Friedman, "Elements of Statistical Learning Ed. 2", Springer, 2009.
 
  .. [R2007] G. Ridgeway, "Generalized Boosted Models: A guide to the gbm package", 2007
+
+
+.. _voting_classifier:
+
+VotingClassifier
+========================
+
+The idea behind the voting classifier implementation is to combine conceptually different machine learning classifiers and use a majority vote or the average predicted probabilities (soft vote) to predict the class labels. Such a classifier can be useful for a set of equally well performing model in order to balance out their individual weaknesses.
+
+
+
+Majority Class Labels (Majority/Hard Voting) 
+--------------------------------------------
+
+In majority voting, the predicted class label for a particular sample is the class label that represents the majority (mode) of the class labels predicted by each individual classifier. 
+
+E.g., if the prediction for a given sample is
+
+- classifier 1 -> class 1
+- classifier 2 -> class 1
+- classifier 3 -> class 2
+
+the VotingClassifier (with ``voting='hard'``) would classify the sample as "class 1" based on the majority class label.
+
+Usage
+..........................
+
+The following example shows how to fit the majority rule classifier:
+
+    >>> from sklearn import datasets
+    >>> from sklearn import cross_validation
+    >>> from sklearn.linear_model import LogisticRegression
+    >>> from sklearn.naive_bayes import GaussianNB 
+    >>> from sklearn.ensemble import RandomForestClassifier
+    >>> from sklearn.ensemble import VotingClassifier
+    >>> import numpy as np
+
+    >>> np.random.seed(123)
```

Comment:
It would probably be preferred to just set the seed everywhere so that the test is side-effect free.